rsa/prime.py

```diff
@@ -69,10 +69,10 @@ def miller_rabin_primality_testing(n, k):
     # Test k witnesses.
     for _ in range(k):
         # Generate random integer a, where 2 <= a <= (n - 2)
-        a = rsa.randnum.randint(n - 4) + 2
+        a = rsa.randnum.randint(n - 3) + 1
 
         x = pow(a, d, n)
-        if x == 1 or x == n - 1:
+        if x in {1, n - 1}:
```

Comment:
Please don't mix code style changes with actual bugfixes. Constructing a set for every test is not a good idea, and might be quite a bit slower than simply having `x == 1 or x == n - 1`.